bin/mkdevbranch

```diff
@@ -127,14 +146,26 @@ rev="dev tag $target"
 [ "$1" ] && who="$1"
 [ "$2" ] && target="${target%.dev*}-$2"
 
+function check_svn_path_exists() {
+    local __resultvar=$1
+    local __path=$2
+    local myresult
+
+    svn info "${__path}" >/dev/null 2>&1
+    myresult=$?
+
+    eval $__resultvar="'$myresult'"
+}
+
 function mksvndir() {
     who=$1
```

Comment:
Actually, yes!  The point of the `eval` on line 157 here is to act as a return value.  This `eval` acts in the same way that a call stack for C would - it assigns the "returned" result into the caller-scoped variable identified by `$__resultvar`.  (Since bash doesn't natively support the concept of functions that return values, you wind up having to pass in the variable you want to hold the result as a function argument instead.)

So, when you call the function:
```bash
check_svn_path_exists exists "svn_pathname"
```
the function will run the SVN check for that path, and then populate `exists` with either a 0 or a 1.

Reply:
Sorry, I meant to ask about the `who=$1`  on line 161, wondering if it should be `local` to avoid risk of shadowing the global

Thank you for the explanation re: `resultvar`, that's a good trick/tool